spec/logstash/plugin_mixins/ecs_compatibility_support/target_check_spec.rb

```diff
@@ -0,0 +1,67 @@
+# encoding: utf-8
+
+require "logstash-core"
+
+require 'logstash/inputs/base'
+require 'logstash/filters/base'
+require 'logstash/codecs/base'
+require 'logstash/outputs/base'
+
+require "logstash/plugin_mixins/ecs_compatibility_support/target_check"
+
+describe LogStash::PluginMixins::ECSCompatibilitySupport::TargetCheck do
+
+  describe "check_target_set_in_ecs_mode" do
+
+    context 'with a plugin' do
+
+      subject(:plugin_class) do
+        Class.new(LogStash::Filters::Base) do
+          include LogStash::PluginMixins::ECSCompatibilitySupport
+          include LogStash::PluginMixins::ECSCompatibilitySupport::TargetCheck
+
+          config :target, :validate => :string
+
+          def register; 42 end
+
+        end
+      end
+
+      it 'skips check when ECS disabled' do
+        plugin = plugin_class.new('ecs_compatibility' => 'disabled')
+        allow( plugin.logger ).to receive(:info)
+        expect( plugin.register ).to eql 42
+        expect( plugin.logger ).to_not have_received(:info)
```

Comment:
This is a bit fragile -- if _any_ other info-level logging gets added to plugin registration in LS core, these specs will break. Let's constrain it to the message we are looking for:

```suggestion
        expect( plugin.logger ).to_not have_received(:info).with(a_string_including "`target` option")
```